### PR TITLE
 Use chunked reader if content market with STREAMING-AWS4-HMAC-SHA256-PAYLOAD header

### LINKED ISSUE #1009: PutObject: chunk-signature in saved object
When using `X-Amz-Content-Sha256: STREAMING-AWS4-HMAC-SHA256-PAYLOAD` without `Content-Encoding: aws-chunked`,  chunk-signature are saved along with the data.
The [documentation](https://docs.aws.amazon.com/pdfs/AmazonS3/latest/API/s3-api.pdf) does not contain a clear answer about the necessity of specify `aws-chunked` when using `X-Amz-Content-Sha256: STREAMING-AWS4-HMAC-SHA256-PAYLOAD`, but AWS saves the object without chunk-signature.

## Steps to Reproduce
```
wget https://github.com/minio/warp/releases/download/v1.0.2/warp_Linux_x86_64.deb
sudo dpkg -i warp_Linux_x86_64.deb
warp run get.yml
```
[get.yml](https://github.com/user-attachments/files/17127961/get.yml.txt)

### Put request tcpdump:
```
PUT /wrong-obj-size-013/wZ7WtxcV/2.1aUIX4oNFA829dRQO.rnd HTTP/1.1
Host: *SENSETIVE-DATA-REMOVED*:9080
User-Agent: MinIO (linux; amd64) minio-go/v7.0.75 warp/1.0.2
Content-Length: 1198
Authorization: AWS4-HMAC-SHA256 Credential=*SENSETIVE-DATA-REMOVED*/20240924/us-east-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date;x-amz-decoded-content-length,Signature=f07685a33fbfaf79dda28d88432627a813b7c9c0b9bf8fee2445c487870c4818
Content-Type: application/octet-stream
X-Amz-Content-Sha256: STREAMING-AWS4-HMAC-SHA256-PAYLOAD
X-Amz-Date: 20240924T142337Z
X-Amz-Decoded-Content-Length: 1024

400;chunk-signature=8aaff24fa81812955f29dcb1e18d522837b3e1e21f8fd1857a09b10d525cc554
ûíÅ¥X(ZKÍ3[W7ªêfÍ
*SOME-DATA-REMOVED*
%y¥þªé«³ÇÁo
0;chunk-signature=cbe50b600993516e6acbfc73d70ab45a0e77c99d5249fc5a7d7d118bd8cd4b1c
```

### Object contents:
```
$ cat 2.1aUIX4oNFA829dRQO.rnd
400;chunk-signature=8aaff24fa81812955f29dcb1e18d522837b3e1e21f8fd1857a09b10d525cc554
ûíÅ¥X(ZKÍ3[W7ªêfÍ
*SOME-DATA-REMOVED*
%y¥þªé«³ÇÁo
0;chunk-signature=cbe50b600993516e6acbfc73d70ab45a0e77c99d5249fc5a7d7d118bd8cd4b1c
```

### PR DESCRIPTION
**Pull request by smallhive**
Closes #1009.

### CODE REVIEW

File: api/auth/center.go [hunk 1]
```diff
@@ -209,7 +211,9 @@ func (c *center) Authenticate(r *http.Request) (*Box, error) {
 		return nil, err
 	}
 
-	if hdr := r.Header.Get(ContentEncodingHdr); hdr == ContentEncodingAwsChunked {
+	amzContent := r.Header.Get(AmzContentSha256)
+
+	if hdr := r.Header.Get(ContentEncodingHdr); hdr == ContentEncodingAwsChunked || amzContent == ContentEncodingChunked {
```
Comment (roman-khimov) [inline api/auth/center.go:216]: I'd change `hdr` here to something more meaningful then (it's about content encoding).

**Review by roman-khimov — COMMENTED** [review]
Please provide an appropriate commit message (short header, some description in the message).

**Comment by roman-khimov** [comment]
> Please provide an appropriate commit message (short header, some description in the message).

And two commits are not really needed here.

**Comment by smallhive** [comment]
Updated

**Review by EESergey — APPROVED** [review]

**Pull request merged**